client/src/modules/vouchers/toolkit/support_patient.js

```diff
@@ -0,0 +1,189 @@
+angular.module('bhima.controllers')
+.controller('SupportPatientKitController', SupportPatientKitController);
+
+// DI definition
+SupportPatientKitController.$inject = [
+  '$uibModalInstance', 'NotifyService', 'CashboxService', 'SessionService', 'data', 'bhConstants',
+  'DebtorService', 'PatientInvoiceService'
+];
+
+// Import transaction rows for a Support Patient
+function SupportPatientKitController(Instance, Notify, Cashbox, Session, Data, bhConstants, Debtors, Invoices) {
+  var vm = this;
+
+  // global variables
+  vm.enterprise = Session.enterprise;
+  vm.gridOptions = {};
+  vm.tool = Data;
+  vm.patientInvoice = false;
+
+  var MAX_DECIMAL_PRECISION = bhConstants = bhConstants.precision.MAX_DECIMAL_PRECISION;
+
+  // expose to the view
+  vm.selectPatientInvoices = selectPatientInvoices;
+  vm.close = Instance.close;
+  vm.import = submit;
+  vm.loadInvoice = loadInvoice;
+  vm.onSelectAccount = onSelectAccount;
+
+  // debtors from store
+  Debtors.store()
+  .then(function (data) {
+    vm.debtors = data;
+  })
+  .catch(Notify.handleError);
+
+  //  optimization with `Store` will be well
+  Invoices.read()
```

Comment:
If so, I propose that you load these in the `loadInvoices()` callback, using the `debtor_uuid` that comes from the patient.  In other words,
```js
function loadInvoices(patient) {
  Invoices.read({ debtor_uuid : patient.debtor_uuid })
    .then(function (invoices) { 
      // bind grid data
      // do something with the invoices
    });
}
```

This will make the module much more lightweight.